torch/optim/rmsprop.py

```diff
@@ -49,7 +67,20 @@ def step(self, closure=None):
                     grad = grad.add(group['weight_decay'], p.data)
 
                 square_avg.mul_(alpha).addcmul_(1 - alpha, grad, grad)
-                avg = square_avg.sqrt().add_(group['eps'])
-                p.data.addcdiv_(-group['lr'], grad, avg)
+
+                avg = None
+                if group['centered']:
+                    grad_avg = state['grad_avg']
+                    grad_avg.mul_(alpha).add_(1 - alpha, grad)
+                    avg = square_avg.addcmul(-1, grad_avg, grad_avg).sqrt().add_(group['eps'])
+                else:
+                    avg = square_avg.sqrt().add_(group['eps'])
+
+                if group['momentum'] > 0:
+                    mom = state['momentum']
+                    mom.mul_(group['momentum']).addcdiv_(group['lr'], grad, avg)
```

Comment:
Applying the LR to the momentum as is done here is equivalent to moving the multiply outside those terms (and holds for Nesterov momentum as well), though I'm not sure if floating-point precision becomes an issue for small learning rates.